### PR TITLE
Build: Benchmarking reporting in PR description

### PR DESCRIPTION
**Pull request by ndelangen**
## What I did

Add benchmarking data at special section in PR descriptions.
Opt-out by not having the section:
`<  !-- BENCHMARK_SECTION -->`
(remove it from the PR body section)

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

| name | before | after | diff | % |
| ---- | ------ | ----- | ---- | - |
| createTime|  23.3s | 7.8s | -15s -482ms | 🔰-197.2% |
| generateTime|  20.4s | 22.3s | 1.8s | 🔺8.3% |
| initTime|  22.3s | 23.4s | 1.1s | 4.9% |
| createSize|  0 B | 0 B | 0 B | NaN% |
| generateSize|  76.5 MB | 76.5 MB | 216 B | 0% |
| initSize|  205 MB | 205 MB | 216 B | 0% |
| diffSize|  128 MB | 128 MB | 0 B | 0% |
| buildTime|  12.8s | 12s | -774ms | 🔰-6.4% |
| buildSize|  7.59 MB | 7.59 MB | 0 B | 0% |
| buildSbAddonsSize|  1.63 MB | 1.63 MB | 0 B | 0% |
| buildSbCommonSize|  195 kB | 195 kB | 0 B | 0% |
| buildSbManagerSize|  2.3 MB | 2.3 MB | 0 B | 0% |
| buildSbPreviewSize|  349 kB | 349 kB | 0 B | 0% |
| buildStaticSize|  0 B | 0 B | 0 B | NaN% |
| buildPrebuildSize|  4.47 MB | 4.47 MB | 0 B | 0% |
| buildPreviewSize|  3.12 MB | 3.12 MB | 0 B | 0% |
| testBuildTime|  0ms | 0ms | 0ms | NaN% |
| testBuildSize|  0 B | 0 B | 0 B | NaN% |
| testBuildSbAddonsSize|  0 B | 0 B | 0 B | NaN% |
| testBuildSbCommonSize|  0 B | 0 B | 0 B | NaN% |
| testBuildSbManagerSize|  0 B | 0 B | 0 B | NaN% |
| testBuildSbPreviewSize|  0 B | 0 B | 0 B | NaN% |
| testBuildStaticSize|  0 B | 0 B | 0 B | NaN% |
| testBuildPrebuildSize|  0 B | 0 B | 0 B | NaN% |
| testBuildPreviewSize|  0 B | 0 B | 0 B | NaN% |
| devPreviewResponsive|  11s | 9.8s | -1s -249ms | 🔰-12.7% |
| devManagerResponsive|  6.9s | 7.2s | 222ms | 3.1% |
| devManagerHeaderVisible|  1.1s | 799ms | -325ms | 🔰-40.7% |
| devManagerIndexVisible|  1.1s | 835ms | -333ms | 🔰-39.9% |
| devStoryVisibleUncached|  1.6s | 1.4s | -129ms | 🔰-8.8% |
| devStoryVisible|  1.1s | 855ms | -327ms | 🔰-38.2% |
| devAutodocsVisible|  961ms | 772ms | -189ms | 🔰-24.5% |
| devMDXVisible|  909ms | 875ms | -34ms | -3.9% |
| buildManagerHeaderVisible|  1s | 742ms | -258ms | 🔰-34.8% |
| buildManagerIndexVisible|  1s | 744ms | -258ms | 🔰-34.7% |
| buildStoryVisible|  1s | 791ms | -275ms | 🔰-34.8% |
| buildAutodocsVisible|  994ms | 686ms | -308ms | 🔰-44.9% |
| buildMDXVisible|  882ms | 700ms | -182ms | 🔰-26% |

### CODE REVIEW

**Comment by ndelangen** [comment]
Please review @kasperpeulen 
Do not merge yet, I need to change back `pull_request` to `pull_request_target` before the merge.

**Review by kasperpeulen — APPROVED** [review]
LGTM!

**Pull request merged**
